src/SFA.DAS.Tools.Servicebus.Support.Web/App_Start/IoC.cs

```diff
@@ -77,7 +78,16 @@ public static IServiceCollection AddServices(this IServiceCollection services, I
             services.AddTransient<ICommandHandler<UpsertUserSessionCommand, UpsertUserSessionCommandResponse>, UpsertUserSessionCommandHandler>();
             services.AddTransient<ICommandHandler<DeleteUserSessionCommand, DeleteUserSessionCommandResponse>, DeleteUserSessionCommandHandler>();
             services.AddTransient<IBatchMessageStrategy, BatchMessageStrategy>();
-            services.AddTransient<IUserSessionService, UserSessionService>();
+            services.AddTransient<IUserSessionService, UserSessionService>(s =>
+                new UserSessionService(
+                    s.GetService<ICommandHandler<UpsertUserSessionCommand, UpsertUserSessionCommandResponse>>(),
```

Comment:
we should be using GetRequiredService rather than GetService so that an exception is thrown if no instance of the type is registered